uat/src/main/java/com.aws.greengrass/FileSteps.java

```diff
@@ -77,19 +95,32 @@ private void createFileAndWriteData(Path tempDirectoryPath, String fileNamePrefi
         }
     }
 
-    private static List<String> generateRandomMessages(int n, int length) {
-        List<String> msgs = new ArrayList<>();
-        for (int i = 0; i < n; i++) {
-            // TODO: Improves this as this is not how the logger writes the logs
-            msgs.add(RANDOM_STRING_GENERATOR.generate(length));
-        }
-        return msgs;
-    }
-
     private void addDataToFile(String data, Path filePath) throws IOException {
         try (BufferedWriter writer = Files.newBufferedWriter(filePath, StandardOpenOption.APPEND)) {
             writer.write(data + "\r\n");
         }
     }
+    @And("I verify the rotated files are deleted except for the active log file for component {word}")
+    public void verifyActiveFile(String componentName) {
+        Path logsDirectory = testContext.installRoot().resolve("logs");
+        if (!platform.files().exists(logsDirectory)) {
+            throw new IllegalStateException("No logs directory");
+        }
+        List<File> componentFiles = Arrays.stream(logsDirectory.toFile().listFiles())
+                .filter(File::isFile)
+                .sorted(Comparator.comparingLong(File::lastModified))
+                .collect(Collectors.toList());
+        assertEquals(1, componentFiles.size());
+        File activeFile = componentFiles.get(componentFiles.size() - 1);
+        // When writing the files for a component store on the scenario context the path of the last file that got
+        // written
+        String expectedActiveFilePath = scenarioContext.get(componentName + "ActiveFile");
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem:**
 This line of code lacks validation when processing input data through the following parameter: &#39;componentName&#39; (index: 0 | type: String). The parameter is exposed to external callers, because its enclosing class and method are publicly accessible. This means that upstream validation, if it exists, can be bypassed. Other methods validating the parameter: &#39;arrangeComponentLogFiles&#39;. The same parameter type is validated here for example: aws-greengrass-log-manager/uat/src/main/java/com.aws.greengrass/CloudWatchSteps.java:91. Malicious, malformed, or unbounded inputs can cause unexpected runtime behavior or crashes, and can slow performance.

**Fix:**
 Add checks to ensure the validity of the parameter&#39;s value, such as testing it for nullness (for example, using the @NonNull annotation as [described in the Lombok library](https://projectlombok.org/features/NonNull)), emptiness, or equality. Or to prevent direct calls to it, [reduce the method&#39;s visibility](https://docs.oracle.com/javase/tutorial/java/javaOO/accesscontrol.html).

 Learn more about potential threats and guidance from the [Common Weakness Enumeration website](https://cwe.mitre.org/data/definitions/693.html) and the [OWASP Cheat Sheet series](https://cheatsheetseries.owasp.org/cheatsheets/Input_Validation_Cheat_Sheet.html).